grails-app/services/org/broadinstitute/orsp/UserService.groovy

```diff
@@ -151,4 +154,57 @@ class UserService {
         }
     }
 
+    /**
+     * Edits a user's roles
+     *
+     * @param userId        The user's id
+     * @param rolesToAssign Roles String array to be assigned
+     */
+    void editUserRoles (Integer userId, ArrayList<String> rolesToAssign) throws SQLException {
+        if (userId != null) {
+            User user = User.findById(userId)
+            def validatedRoles = validateRoles(rolesToAssign)
+            if (user != null && !validatedRoles?.hasErrors) {
+                queryService.deleteOrspUserRoles(userId)
+                if (!rolesToAssign.isEmpty()) {
+                    queryService.updateOrspUserRoles(user, rolesToAssign)
+                }
+            } else {
+                log.error("Error while trying to modify roles to userId: ${userId}. ${validatedRoles.errorMessage}")
+                throw new IllegalArgumentException()
```

Comment:
nit: suggestion for what to add to this exception: `"Error while trying to modify roles for user ${user.displayName}"`

Reply:
Sounds good! I used the message that you suggested.